client/client.go

```diff
@@ -35,10 +35,14 @@ import (
 	"github.com/snapcore/snapd/dirs"
 )
 
-func unixDialer(_, _ string) (net.Conn, error) {
+func privateUnixDialer(_, _ string) (net.Conn, error) {
```

Comment:
The private and public analogy is perhaps not the best one for this purpose. Neither of these sockets or the APIs accessible through them are quite private or public. One of them is accessible by the whole system and by snaps carrying snapd-control interface, and the other is available to any snap.

I suggest naming the two sockets:
```
snapd.socket
snap.socket
```
and following that idea in the rest of the terminology. I'll follow that below.